src/nwbuild.js

```diff
@@ -93,27 +101,17 @@ export const nwbuild = async (options) => {
 
     // Get current platform and arch if mode is run
     if (options.mode === "run") {
-      let tmpPlatform = getPlatform(platform);
-      let tmpArch = getArch(arch);
-      if (tmpPlatform === undefined) {
+      if (options.platform === undefined) {
         throw new Error(`Platform ${platform} is not supported. Sorry!`);
       }
-      if (tmpArch === undefined) {
+      if (options.arch === undefined) {
         throw new Error(`Architecture ${arch} is not supported. Sorry!`);
```

Comment:
Why would an error be thrown in "run" mode if either `platform` or `arch` is undefined? Theoretically, because of changes to `parse()`, these would never actually be undefined. However, if a user is attempting to run the app, they would be trying to use the currently running OS. So, shouldn't `platform` and `arch` **always** match the current OS and architecture in "run" mode, even if something different is specified in `package.json` or the `nw-builder` configuration object?

Reply:
Platform, yes. But on Windows 64, it is more common to run 32-bit software than 64-bit. So they may intentionally want to run a 32-bit version of NW.js to ensure that all dependencies work correctly.

Reply:
> Platform, yes. But on Windows 64, it is more common to run 32-bit software than 64-bit.

This is simply not true and I don't understand why you keep trying to propagate it. It is days away from the year 2023. It is absolutely not more common to be running 32-bit and should **not** be done in most cases. Why would you limit your apps memory and other functionality by default, when 64-bit has been the standard for nearly a decade?

Reply:
Though very uncommon to be running a 32-bit *OS*, most Windows software that is installed today is still 32-bit. This is pretty easy to check just by comparing the amount of items in `C:\Program Files` versus `C:\Program Files (x86)`. It's of course going to depend on the user, but across my machines about 65% of the software is still 32-bit.

> Why would you limit your apps memory and other functionality by default?

None of my apps require much resources. If I was making software that taxed the hardware, I'd offer a 64-bit version of my app for Windows. Or if my app relies on a dependency that only supports 64-bit, then I'd be locked out of supporting 32-bit. But otherwise, if my app does not benefit from 64-bit, then I use 32 by default. It is functionally no different to me or my users and will work on more systems and I have one less build to do. Sure there aren't many 32-bit Windows users out there, but realistically there are likely more rare 32-bit Windows users than all non-windows users combined. I use NW.js to get wider platform support for free. If there's no downside to how my app runs, then I'll publish it in the most universal way possible to support the most platforms.

You should really be asking yourself, what benefit do you get by excluding this subset of users. If it requires additional effort, then by all means, skip them, but for most cases, it's the same level of effort to support 32 & 64 bit Windows as it is to only support 64-bit. So might as well do both. For example, I made an app a year or two ago, and it required me to build a native dependency. I decided that until someone requested a 32-bit build, that I wouldn't put forth the effort to support both, as no one would likely request it. I instead put that effort towards finding another native dependency that worked on OSX, building, testing, and releasing a version for that OS (that then no one downloaded). But as a side effect it worked for Linux too, and nearly half of my downloads are for Linux now. So I guess it worked out in the end.

Reply:
> Why would an error be thrown in "run" mode if either platform or arch is undefined?

Before adding in error handling, `undefined` conveyed that the platform/arch is not supported. For example, if a Linux arm64 user downloads Linux amd64 binary, trying to run it would fail with an unsupported arch message.

Currently, it looks like these functions have become redundant. Instead, the validate function should check for platform-arch support.

Reply:
My two cents:
- For "run" mode, use the current OS for `platform` no matter what's in `package.json`. For `arch`, you both make a good point about needing to fail if a user's running Linux arm64. Might want to consider automatically failing over from arm64 to x64 for osx, until a native build is available.
- For any "build" modes, I'm clearly out-numbered but still think defaults should be x64 unless otherwise specified. It'll never make sense to me to sacrifice performance/features to support a tiny fragment of potential users. If 32-bit support is actually needed by your users, by all means support it. Otherwise, it's time to move forward. Let's stop contributing to the `Program Files (x86)` content...